src/buffer/out/textBuffer.cpp

```diff
@@ -1028,3 +1028,243 @@ const TextBuffer::TextAndColor TextBuffer::GetTextForClipboard(const bool lineSe
 
     return data;
 }
+
+// Routine Description:
+// - Generates a CF_HTML compliant structure based on the passed in text and color data
+// Arguments:
+// - rows - the text and color data we will format & encapsulate
+// - iFontHeightPoints - the unscaled font height
+// - fontFaceName - the name of the font used
+// Return Value:
+// - string containing the generated HTML
+std::string TextBuffer::GenHTML(const TextAndColor& rows, const int iFontHeightPoints, const PCWCHAR fontFaceName)
+{
+    std::string szClipboard; // we will build the data going back in this string buffer
+
+    try
+    {
+        std::string const szHtmlClipFormat =
+            "Version:0.9\r\n"
+            "StartHTML:%010d\r\n"
+            "EndHTML:%010d\r\n"
+            "StartFragment:%010d\r\n"
+            "EndFragment:%010d\r\n"
+            "StartSelection:%010d\r\n"
+            "EndSelection:%010d\r\n";
+
+        // measure clip header
+        size_t const cbHeader = 157; // when formats are expanded, there will be 157 bytes in the header.
+
+        std::string const szHtmlHeader =
+            "<!DOCTYPE><HTML><HEAD><TITLE>Windows Console Host</TITLE></HEAD><BODY>";
+        size_t const cbHtmlHeader = szHtmlHeader.size();
+
+        std::string const szHtmlFragStart = "<!--StartFragment -->";
+        std::string const szHtmlFragEnd = "<!--EndFragment -->";

```

Comment:
This constant is only used once (in line 1239), its definition should be moved to that location and the constant should be changed to `constexpr std::string_view`:

```
        // copy HTML end fragment
        constexpr std::string_view szHtmlFragEnd = "...";
        szClipboard.append(szHtmlFragEnd);
```

Given that the constant is only used once and the point of use even has a comment stating what is being done (*append end fragment*) you can even consider removing the constant altogether, as this code is not less readable than the above:

```
        // copy HTML end fragment
        szClipboard.append("...");
```